### ISSUE TITLE
Use exec for applications spawned via shell. (Enhancement)

### ISSUE THREAD

**Issue by gnarendran** [ranger/ranger#1318]
#### Current Behavior
When spawning an external program from rifle.conf, the shell is invoked and this shell needlessly stays alive (can be seen in ps) till the program exits.

For example when vlc is spawned by ranger ps shows both:
/bin/sh -c set -- '/mnt/Music/PLAYLIST_ALL.xspf'; vlc -- "$@"
and
vlc -- /mnt//Music/PLAYLIST_ALL.xspf

#### Context
In htop output, noticed many of these redundant shells simply waiting for the external programs to die. They do consume some resources and could be an issue if present in large numbers.

#### Possible Solutions
1. Currently working around in rifle.conf by adding an exec before every program, for example:
`ext xspf, has vlc, X, flag f = exec vlc -- "$@" `                                 

2. It would be cleaner to add **exec** when building the command. For example, in ranger/ext/rifle.py,  "_build_command" could:
`return "set -- '%s'; exec %s" % (filenames, action)`
instead of the current:
`return "set -- '%s'; %s" % (filenames, action)`

And similarly in ranger/core/fm.py sxiv_workaround_hook.

**Comment by gnarendran** [comment]
Thanks for the quick response! I Agree solution 1 (unlike 2) allows for "action" to be a compound shell command, and so is preferable.

**Comment by toonn** [comment]
Maybe this should be a rifle flag?

**Comment by Vifon** [comment]
No need for a flag as specifying "exec" before the command would be functionally equivalent.

**Comment by toonn** [comment]
I just figured it'd be more declarative and consistent with `flag f`. Fork's always treated differently/implicitly but I don't see a good reason, the syscalls are similar enough. Not a big deal either way.

**Comment by Vifon** [comment]
Flags are great for things immediately noticeable by the user. `exec` is more of an optimization / being "clean" than a real change. I cannot imaging a case when a user would enable this flag ad hoc like for example the new terminal flag.

**Comment by toonn** [comment]
Exactly, that's why it'd be a default and you use the no-exec flag : p for those rare occasions where you don't want it : )
